sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -175,6 +175,59 @@ object GpuCast extends Arm {
       cv.stringReplaceWithBackrefs(rule.search, rule.replace)
     })
   }
+
+  def sanitizeStringToIntegralType(input: ColumnVector, ansiEnabled: Boolean): ColumnVector = {
+    // Convert any strings containing whitespace to null values. The input is assumed to already
+    // have been stripped of leading and trailing whitespace
+    val sanitized = withResource(GpuScalar.from(null, DataTypes.StringType)) { nullVal =>
+      withResource(input.containsRe("\\s")) { hasWhitespace =>
+        withResource(hasWhitespace.any()) { any =>
+          if (any.getBoolean) {
+            if (ansiEnabled) {
+              throw new NumberFormatException(GpuCast.INVALID_INPUT_MESSAGE)
+            } else {
+              hasWhitespace.ifElse(nullVal, input)
+            }
+          } else {
+            input.incRefCount()
+          }
+        }
+      }
+    }
+
+    // truncate strings that represent decimals, so that we just look at the string before the dot
+    if (ansiEnabled) {
+      // ansi mode does not support casting decimal strings to integers
+      sanitized
+    } else {
+      withResource(sanitized) { _ =>
+        withResource(Scalar.fromString(".")) { dot =>
+          withResource(sanitized.stringContains(dot)) { hasDot =>
+            // only do the decimal sanitization if any strings do contain dot
+            withResource(hasDot.any(DType.BOOL8)) { anyDot =>
+              if (anyDot.getBoolean) {
+                // Special handling for strings that have no numeric value before the dot, such
+                // as "." and ".1" because extractsRe returns null for the capture group
+                // for these values and it also returns null for invalid inputs so we need this
+                // explicit check
+                withResource(sanitized.matchesRe("^[+\\-]?\\.[0-9]*$")) { startsWithDot =>
+                  withResource(sanitized.extractRe("^([+\\-]?[0-9]*)\\.([0-9]*)?$")) { table =>
```

Comment:
It looks like there are two capture groups here, but we're only looking at one of them.  It would be nice to avoid manifesting the second group if we don't need it.